sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2747,6 +2747,17 @@ class Dataset[T] private[sql](
     }
   }
 
+  /**
+   * Collect a Dataset as ArrowPayload byte arrays and serve to PySpark.
+   */
+  private[sql] def collectAsArrowToPython(): Int = {
+    val payloadRdd = toArrowPayloadBytes()
+    val payloadByteArrays = payloadRdd.collect()
```

Comment:
Ok. As actually to make the Pandas's DataFrame, all data are needed to load into driver's memory, `toLocalIteratorAndServe` can't improve the memory usage in the end.